### PR TITLE
lsp-csharp.el: add handler for the "omnisharp/client/findReferences"

### PR DESCRIPTION
**Pull request by razzmatazz**
This will make it possible to use code lenses to find symbol references.

Previously only reference count was displayed but now it possible to invoke action on the references element too (via `lsp-avy-lens`)

The related PR on server side is:
 - https://github.com/OmniSharp/omnisharp-roslyn/pull/1807

### CODE REVIEW

File: lsp-csharp.el [hunk 1]
```diff
@@ -244,6 +244,21 @@ tarball or a zip file (based on a current platform) to TARGET-DIR."
   "Resolves path and arguments to use to start the server."
   (list (lsp-csharp--language-server-path) "-lsp"))
 
+(defun lsp-csharp--action-client-find-references (action)
+  "Reads first argument from ACTION as Location and displays xrefs for that location
+using the `textDocument/references' request."
+  (if-let ((arguments (gethash "arguments" action))
```
Comment (yyoncho) [inline lsp-csharp.el:250]: can you rewrite it using lsp-protocol.el bindings?

Comment (razzmatazz) [reply lsp-csharp.el:250]: I am not entirely sure.. Can you point me to any example of the code that does similar thing? 

I do need to define schema for this particular command using lsp-protocol.el, right?

Comment (yyoncho) [reply lsp-csharp.el:250]: AFACS there are existing types - Command, Location should be sufficient. Check this:

https://github.com/emacs-lsp/lsp-mode/blob/plists/lsp-mode.el#L2316

Comment (razzmatazz) [reply lsp-csharp.el:250]: I am not sure I understand your pointer to `lsp--lens-create-interactive-command`.. that function returns a lambda that invokes a command with command and args..

But.. do you mean, I need to decorate (with a macro presumably?) `lsp-csharp--action-client-find-references` function so `lsp-protocol.el` could somehow parse the conents of `action` into subcomponents that i actually need (and currently extract within if-let expression?)

Comment (kiennq) [reply lsp-csharp.el:250]: It should be something like this
```suggestion
  (-if-let* (((&Command: arguments?) action))
```

Or you can even change the `defun` to `lsp-defun` and
``` elisp
(lsp-defun lsp-csharp--action-client-find-references ((&Command :arguments?))
```

Comment (razzmatazz) [reply lsp-csharp.el]: I have updated the code using -if-let* / destructuring. Sorry for the long delay

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
> clients/lsp-csharp.el:238:1:Error: Unused lexical variable ‘command’

There is one warning which is breaking the CI, can you fix it?

**Comment by razzmatazz** [comment]
> > clients/lsp-csharp.el:238:1:Error: Unused lexical variable ‘command’
> 
> There is one warning which is breaking the CI, can you fix it?

Done

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**